### PR TITLE
Solution for #Robot_factory

### PR DESCRIPTION
**Pull request by jstmpelowycz**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,75 @@
...
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    this.name = name || 'unknown';
+    this.weight = weight || 0;
+    this.chipVersion = chipVersion || 0;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0, y: coords.y || 0,
```
Comment (SanyaBratashchuk) [inline src/robotFactory.js:10]:
```suggestion
      x: coords.x || 0, 
      y: coords.y || 0,
```

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,75 @@
...
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name, weight, coords = {},
+    chipVersion, maxLoadWeight, currentLoad
+  ) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight || 0;
+    this.currentLoad = currentLoad || null;
```
Comment (SanyaBratashchuk) [inline src/robotFactory.js:60]:
```suggestion
    this.maxLoadWeight = maxLoadWeight
    this.currentLoad = currentLoad
```

File: src/robotFactory.js [hunk 3]
```diff
@@ -1,10 +1,75 @@
...
+  }
+
+  hookLoad(cargo = {}) {
+    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
+      this.currentLoad = { ...cargo };
+    }
```
Comment (SanyaBratashchuk) [inline src/robotFactory.js:66]:
```suggestion
  hookLoad(cargo) {
    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
      this.currentLoad = cargo;
    }
```

**Review by SanyaBratashchuk — APPROVED** [review]

File: src/robotFactory.js (hunk 3 above)
Comment (jstmpelowycz) [reply src/robotFactory.js:66]: agree

File: src/robotFactory.js (hunk 2 above)
Comment (jstmpelowycz) [reply src/robotFactory.js:60]: why not to provide any kind of finely saved items?)

File: src/robotFactory.js (hunk 1 above)
Comment (jstmpelowycz) [reply src/robotFactory.js:10]: agree

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**